### DISCUSSION TITLE
welp we see how things do tonight

### DISCUSSION THREAD

**Discussion by nealmick** [nealmick/bb#40] [General]
i think i build last game 3 soon.  last game 2 seems to have helped. then i need to start thinking about middle data, in between the short term momentum and long term season averages.... seems like things are doing a bit better, like there are sparks of it working.  I am starting to only push new features to live server.....  I plan on still support local installs.  I think the overall idea that we have a base install that gets most things you already have now, and if you really want the cool new features then use live server.... that way i have a bit of a gap there.... and somtimes i may make big release with lots feature.   
also notice change if you train with or without spread... like without spread you get some wild predictions, but they may be more accurate.  And with spread model kinda follows what the spread does to some degree.... soo yea recommend trying models with and without spread...  ohhh yea i want to build out drop all players.... ooo ill just make the slider go all the way to 0... then we can have model that dont dont use the top season average table, and then you could even not worry about removing injured players.... because it would just be team level stats and last game data.... not season averages with changing rosters......
okay i watch borat tonight.

**Comment by Wolf242** [comment]
Also, maybe add where you can select which players may be questionable to play for an injury. That way the model can calculate that the player may not play at 100% returning from injury so the player may end up with maybe reduced minutes.

**Reply by nealmick** [reply]
a great idea, but this would really difficult to do, okay i think this through and write down my thoughts.  thing is we dont have historical injuries... Like we kinda do, but we infer the rosters based on who actually played..... soo like the top season averages in training are calculated by the actual players that played, thats how we determine the roster.... however when it comes time to predict a future game.... we dont know who will actually play.... no one really does.... and if a player plays like the best data i think for future games is the doctor injury report..... uhm i kinda get what you mean by like players that were recently injured should some how be downgraded.... and i kinda agree with that... im not sure if thats always the case? some players may come back stronger??? also im really not sure how we would adjust the season averages... like we would have to determine the adjustment some how, and that would be needed for bolth historical and future games......   I really think focusing in on last game and recent data is gona be the path forward here.... because if a player has been injured they just arnt in the recent game data... then as the player returns that start showing up on the last game history again with there actual stats from that last game.... soo it would kinda lag by 1 game as players return by its nature its game history...... but we trying to map or generalize how our inputs effect game scores..... sooo yea..... i mean i like the idea and your probably correct that in most cases players with recent injuries play worse, i wouldnt doubt that.  i dont know how we would account for this specifically.... i kind think this like finding a particular data point that you would like, and then figuring out back from there.  I find this does not help me, because you just come up with ideas that are really difficult to build, and possibly there is not sufficient data out there recorded to model what ever the feature is....... ssoooooooo how i think of ideas to move the project forward is looking at commonly available data or data that we have already captured and forming it differently to better represent a game.....    soooooo like we have all the actualy player stats from every game.  and we form that in sense as last game data..... we also have season averages from every season..... soo we combine this data to detirmine a roster of season averages from the players that actually player and then in addition we use the last game data..... and then we also have team level stats win/loss record streak.... and spread and so on....... with all them togled on training dashboard.... i hieghly recommend just trying to train models.... especially with different amount of players and different data settings.... like there are soo many possible combinations there that could lead to the model learning faster and being more accurate....  sparse data is a thing, if your data isnt dense, and potent its more difficult to learn patterns.... because there is noise in the signal.... you want to provide the model with clearest signal indication data as to what will happen.... the model the attempts build up relationships and map how different data points effect the target game scores....... okay i back to watching more borat clips for now. haha
i no even read what i write here i jsut click send

**Reply by nealmick** [reply]
also by the way minutes is adjustable... you can edit player stats before making predictions.... sooo if you want to run a prediction with edited stats thats no problem.  just click on player stat after removing players and before predicting.... like  lets me screen shot this:
![Screen Shot 2023-11-29 at 10 31 57 PM](https://github.com/nealmick/bb/assets/33704976/8cb10101-9369-4c39-8c4b-6c6be7517132)


just edit stats after removing players, because edited stats get cleared when removing players.... editing stats should be the last thing done before predict.....   

like behind the scenes we actually recreate the game when you remove a player, but we recreate the game without that player... soo it clears your edited stats....

**Reply by nealmick** [reply]
btw i like never use this feature sooo like i have no clue man, might be super usefull might not .... it was one of the first thing i built actually and its also why we have that crappy bar, so that its one single table we map your edits to the local file on the server just before predicting..... i call this editable stats i think there is some info in the readme about it too lazy to link haha

**Reply by nealmick** [reply]
also only season averages are editable not last game data, because that game already happened and kinda no point to edit last game stats....

**Reply by nealmick** [reply]
like i really think people dont realise how much complexity and time i put into this damn thing.... like there are soo many hidden features haha

**Reply by nealmick** [reply]
https://www.youtube.com/watch?v=eH9w6kxPIyU

**Reply by Wolf242** [reply]
Nice, I do have some ideas to enhance based off what you mentioned. I'll provide in a new topic.

**Reply by nealmick** [reply]
sounds good

**Reply by nealmick** [reply]
i think i just need to write better docs or somthing??? like from your perspective what can i do make it easier to understand how to use the system??? like if added tour thing onboarding like that annoying shit i always skip and websites.... i could make somthing like that but that seems really annoying, i could make chat thing, and then give gpt the docs and all the info so gpt know how to use then people could chat and ask question and gpt could repsond hopefully informed of how the system works.... there is alot in the docs and there are like question mark icons that have alot of info on the live server too.... the local install is a bit more bare... because i have to be carefull about like the shop and there are difference with the live and local installs, its the exact same thing....  there same project but two different branches lets say.... although the live server is just live... its docent really have its own branch,.... oo this it i have my own local dev branch that is pretty much in sync with the live server.....  idk how i got started on this.  i was watching borat then i remember another thing.,.. idk man... honestly been working on this soo long still docent work soo open to any ideas.... also i think just messing with the system there are soo many options and ways to use it.... there is probably some value here and there... just have to find it..... i dont think vegas odds or predicting future are easy things....... how difficult idk yet to really get the system to point i want it to be at.... like ya know... would be great if it was just magic.... but we 4 years into this project.... i made the first folder of when i started writing stuff for this over 4 years ago.

also i just want to be clear my background and expertise is in information systems. not basket ball, not vegas odds.... i dont bet on games, and am not a gambler.... not even really all that interested in betting.  i just want to predict games.... and hopefully sell that system to other people to bet... but i dont want it to be scam... that why i work so hard to build system.   Like it needs to be profitable to not be scam.... betting has limits. its not money printer.  casinos just tell you no, sorry you can no long place bets, your  a sharp better.... sooo you can make some money but not infinite.....  soo i build system instead, not try to become gambler.... 

okay back to borat
